tests/test_make.py

```diff
@@ -240,8 +240,8 @@ class E(D, C):
             e2 = attr.ib(default="e2")
 
         assert (
-            "E(a1='a1', a2='a2', b1='b1', b2='b2', c1='c1', c2='c2', d1='d1', "
-            "d2='d2', e1='e1', e2='e2')"
+            "E(a1='a1', a2='a2', d1='d1', d2='d2', b1='b1', b2='b2', "
+            "c1='c1', c2='c2', e1='e1', e2='e2')"
```

Comment:
Why was d args following a args???  E is first subclassing from D, which sub-classes A.  So it is only logical and sensible to have A and D args first!  or what am I missing?

Reply:
ah, may be because there is another route to A through C, so expectation was that they would somehow "squeeze" (overload) previously defined ones out?  but then still then `d`s should then lead the list of args.